File: app/src/main/java/com/cyl/music_hnust/dataloaders/db/MusicCursorWrapper.java
package com.cyl.music_hnust.dataloaders.db;

import android.database.Cursor;
import android.database.CursorWrapper;

import com.cyl.music_hnust.model.music.Music;
import com.cyl.music_hnust.model.music.Playlist;

public class MusicCursorWrapper extends CursorWrapper {

    public MusicCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    public Music getMusic() {
        Music music = new Music();
        music.setId(getInt(getColumnIndex(DBData.MUSIC_ID)));
        music.setTitle(getString(getColumnIndex(DBData.MUSIC_NAME)));
        music.setArtist(getString(getColumnIndex(DBData.MUSIC_ARTIST)));
        music.setAlbum(getString(getColumnIndex(DBData.MUSIC_ALBUM)));
        music.setDuration(getLong(getColumnIndex(DBData.MUSIC_TIME)));
        music.setUri(getString(getColumnIndex(DBData.MUSIC_PATH)));
        music.setCoverUri(getString(getColumnIndex(DBData.MUSIC_ALBUM_PATH)));
        music.setFileName(getString(getColumnIndex(DBData.MUSIC_FILENAME)));
        music.setFileSize(getLong(getColumnIndex(DBData.MUSIC_SIZE)));
        music.setYear(getString(getColumnIndex(DBData.MUSIC_YEARS)));
        return music;
    }

    public Playlist getPlaylist() {
        String id = getString(getColumnIndex(DBData.PLAYLIST_ID));
        String name = getString(getColumnIndex(DBData.PLAYLIST_NAME));
        int num = getInt(getColumnIndex("num"));
        return new Playlist(id, name, num);
    }

}

File: app/src/main/java/com/cyl/musiclake/data/db/MusicCursorWrapper.kt
package com.cyl.musiclake.data.db

import android.database.Cursor
import android.database.CursorWrapper

import com.cyl.musiclake.bean.Album
import com.cyl.musiclake.bean.Artist

class MusicCursorWrapper(cursor: Cursor) : CursorWrapper(cursor) {
    
    val album: Album
        get() {
            val id = getString(getColumnIndex("albumid"))
            val name = getString(getColumnIndex("album"))
            val artistId = getLong(getColumnIndex("artistid"))
            val artist = getString(getColumnIndex("artist"))
            val num = getInt(getColumnIndex("num"))
            return Album(id, name, artist, artistId, num)
        }

    val artists: Artist
        get() {
            val id = getLong(getColumnIndex("artistid"))
            val name = getString(getColumnIndex("artist"))
            val num = getInt(getColumnIndex("num"))
            return Artist(id, name, num)
        }

}
